components/brave_new_tab_ui/reducers/rewards_reducer.ts

```diff
@@ -0,0 +1,189 @@
+      state = { ...state }
+      state.rewardsState.adsEstimatedEarnings = payload.amount
+      break
+
+    case types.ON_BALANCE_REPORTS:
+      state = { ...state }
+      const reports = payload.reports || {}
+      state.rewardsState.totalContribution = getTotalContributions(reports)
+      break
+
+    case types.DISMISS_NOTIFICATION:
+      state = { ...state }
+
+      const dismissedNotifications = state.rewardsState.dismissedNotifications
+      dismissedNotifications.push(payload.id)
+      state.rewardsState.dismissedNotifications = dismissedNotifications
+
+      state.rewardsState.grants = state.rewardsState.grants.filter((grant) => {
+        return grant.promotionId !== payload.id
+      })
+      break
+
+    case types.ON_GRANT:
+      state = { ...state }
+
+      if (action.payload.properties.status === 1) {
+        break
+      }
+
+      const promotionId = payload.properties.promotionId
+      if (!promotionId) {
+        break
+      }
+
+      if (!state.rewardsState.dismissedNotifications) {
+        state.rewardsState.dismissedNotifications = []
+      }
+
+      if (state.rewardsState.dismissedNotifications.indexOf(promotionId) > -1) {
+        break
+      }
+
+      const hasGrant = state.rewardsState.grants.find((grant: NewTab.GrantRecord) => {
+        return grant.promotionId === promotionId
+      })
+      if (hasGrant) {
+        break
+      }
+
+      const updatedGrants = state.rewardsState.grants
+      updatedGrants.push({
+        promotionId: promotionId,
+        type: payload.properties.type
+      })
+
+      state.rewardsState.grants = updatedGrants
+      break
+
+    case types.ON_BALANCE:
+      state = { ...state }
+      state.rewardsState.balance = payload.balance
+      break
+
+    case types.ON_WALLET_EXISTS:
+      if (!payload.exists || state.rewardsState.walletCreated) {
+        break
+      }
+      state.rewardsState.walletCreated = true
+      break
+
+    case types.SET_PRE_INITIAL_REWARDS_DATA:
+      const preInitialRewardsDataPayload = payload as PreInitialRewardsData
+      state = {
+        ...state,
+        rewardsState: {
+          ...state.rewardsState,
+          enabledAds: preInitialRewardsDataPayload.enabledAds,
+          enabledMain: preInitialRewardsDataPayload.enabledMain
+        }
+      }
+      break
+
+    case types.SET_INITIAL_REWARDS_DATA:
+      const initialRewardsDataPayload = payload as InitialRewardsData
+      const newRewardsState = {
+        onlyAnonWallet: initialRewardsDataPayload.onlyAnonWallet,
+        balance: initialRewardsDataPayload.balance,
+        totalContribution: getTotalContributions(initialRewardsDataPayload.reports)
+      }
+
+      state = {
+        ...state,
+        rewardsState: {
+          ...state.rewardsState,
+          ...newRewardsState
+        }
+      }
+      break
+
+    default:
+      break
+  }
+
+  if (state !== startingState) {
+    storage.debouncedSave(state)
+  }
```

Comment:
isn't saving the state to storage here redundant since it's happening in mergeReducers?

Reply:
yep, removed